### PR TITLE
Add Advanced/GitOps documentation

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/roadmap/issues/415

### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [x] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/main/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

**Review by piontec — APPROVED** [review]
Well written, I like the tutorial style of this doc

File: src/content/advanced/gitops/index.md [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+        --prune=true  \
+        --interval=30s \
+        --validation=client \
+        --export > 02-organization.yaml
+```
+
+For Flux to manage this piece of code, you just have to create the `Kustomization` resource. Contents of `02-organization` will be reconciled automatically.
+
+```nohighlight
+kubectl create -f 02-organization.yaml
```
[the remark this answers is not in the document]
Comment (kubasobon) [reply src/content/advanced/gitops/index.md:159]: Resources in `flux-demo` are using `v1beta1`. We should upgrade flux in MCs ASAP.

File: src/content/advanced/gitops/index.md [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+> - parameterize common variables and put them in `values.yaml`: cluster ID, organization name, namespace, control plane ID, etc.
+> - add hook annotations to `Cluster` CR: `helm.sh/hook: pre-install` and `helm.sh/hook-weight: "-1"`
+
+The cluster will still need a NodePool to reach full functionality. To learn more, visit [NodePool documentation]({{< relref "/advanced/node-pools/index.md" >}}). We will be creating one using `kubectl gs template` command.
+
+```nohighlight
+kubectl gs template nodepool --provider aws --cluster-name demo0 --description demo0 --organization flux-demo --availability-zones 1 > 03-cluster-aws/templates/nodepool-resources.yaml
+```
+
+Add `Chart.yaml` and `values.yaml` from the example repository or create your own.
```
[the remark this answers is not in the document]
Comment (kubasobon) [reply src/content/advanced/gitops/index.md:229]: Done.

File: src/content/advanced/gitops/index.md [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+> Note: There is some parameterization and hooks already added in the demo repository. You can copy from there if you don't want to do it on your own.
+> Otherwise please:
+>
+> - parameterize common variables and put them in `values.yaml`: cluster ID, organization name, namespace, control plane ID, etc.
+> - add hook annotations to `Cluster` CR: `helm.sh/hook: pre-install` and `helm.sh/hook-weight: "-1"`
+
+The cluster will still need a NodePool to reach full functionality. To learn more, visit [NodePool documentation]({{< relref "/advanced/node-pools/index.md" >}}). We will be creating one using `kubectl gs template` command.
+
+```nohighlight
+kubectl gs template nodepool --provider aws --cluster-name demo0 --description demo0 --organization flux-demo --availability-zones 1 > 03-cluster-aws/templates/nodepool-resources.yaml
```
[the remark this answers is not in the document]
Comment (kubasobon) [reply src/content/advanced/gitops/index.md]: I've updated the command and flux-demo repo.

File: src/content/advanced/gitops/index.md [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+        --export > 04-managed-app.yaml
+```
+
+Install the `Kustomization`:
+
+```nohighlight
+kubectl create -f 04-managed-app.yaml
+kustomization.kustomize.toolkit.fluxcd.io/install-managed-app created
+
+kubectl get apps -n demo0 nginx-ingress-controller-app
```
[the remark this answers is not in the document]
Comment (kubasobon) [reply src/content/advanced/gitops/index.md:325]: Done

**Review by rossf7 — APPROVED** [review]
Nice! LGTM

**Pull request merged**